### PR TITLE
Theme cleanup, Fixed Linting, Added Bracket Highlighting

### PR DESCRIPTION
**Pull request by vpillinger**
Cleaned up the code surrounding theme changes. Created a MainMenu class for the purpose of consolidating Menu related code. Also made the menu interface look a little more professional.

Note: Code Mirror allows multiple themes at once, and I thought about implementing this in the menu, but decided against implementing it for now.

Issues resolved:
https://github.com/wellsjo/JSON-Splora/issues/1

### CODE REVIEW

File: app/system/MainMenu.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		MainMenu.app = app
+		
+		// Add theme sub menu
+		menuTemplate[2].submenu[7].submenu = this.createThemeSubMenu()
+
+		// Finally, build menu
+		Menu.setApplicationMenu(Menu.buildFromTemplate(menuTemplate))
+	}
+
+	/**
```
Comment (wellsjo) [inline app/system/MainMenu.js]: little spacing issue here

File: app/system/MainMenu.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+class MainMenu {
+
+	// Create the main menu for the given app
```
Comment (wellsjo) [inline app/system/MainMenu.js]: please use block / jsdoc-style comments for functions

File: index.html [hunk 3]
```diff
@@ -21,6 +21,8 @@
   <script src="node_modules/codemirror/addon/fold/comment-fold.js" type="text/javascript"></script>
   <script src="node_modules/codemirror/addon/fold/indent-fold.js" type="text/javascript"></script>
   <script src="node_modules/codemirror/mode/javascript/javascript.js" type="text/javascript"></script>
+  <script src="node_modules/codemirror/addon/edit/matchbrackets.js" type="text/javascript"></script>
+
```
Comment (wellsjo) [inline index.html]: extra line here

**Comment by vpillinger** [comment]
I updated the title to reflect what is in my branch now. I incorporated those style fixes. I added bracket matching. I also fixed the linter by registering a helper with CodeMirror. Currently, I am just running the parser and returning the errors if any were found. The thing is that the red underline is either going to be 1 character, or the entire document. The parser isn't really a good linter on its own. However, it is certainly better than no linter.

**Comment by wellsjo** [comment]
Awesome! Checking this out now.  We can fix the colors later 😄

**Comment by wellsjo** [comment]
I fixed a few style things and changed `MainMenu` to use class syntax. Merging for now, the rest can be added later!

File: app/system/MainMenu.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+  /**
+   * Create the main menu for the given app
+   */
+
+  constructor(app) {
+    this.app = app
+
+    // Add theme sub menu
+    menuTemplate[2].submenu[7].submenu = this.createThemeSubMenu()
```
Comment (MaanooAk) [inline app/system/MainMenu.js]: That's not even running!
The `menuTemplate[2]` must be `menuTemplate[1]`.

File: app/system/MainMenu.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      type: 'radio',
+      click: (menuItem, browserWindow, event) => {
+        this.themeClicked(menuItem, browserWindow, event)
+      }
+    }, {
+      label: 'Neat',
+      theme: 'neat',
+      type: 'radio',
+      click: (menuItem, browserWindow, event) => {
+        self.themeClicked(menuItem, browserWindow, event)
```
Comment (MaanooAk) [inline app/system/MainMenu.js]: Here too, `self` to `this`

File: index.html [hunk 6]
```diff
@@ -79,16 +80,10 @@
...
+    const MainMenu = require('./app/system/MainMenu')
     const App = require('./app/js/App')
-    const APP = new App()
-
-    // Create main menu (requires `APP` to be instantiated)
-    const {remote} = require('electron')
-    const {Menu} = remote
-    const menuTemplate = require('./app/system/menu')
-    Menu.setApplicationMenu(Menu.buildFromTemplate(menuTemplate))
+    const app = new App()
```
Comment (MaanooAk) [inline index.html]: You renamed `APP` to `app`, can you rename also the refs at `menuTemplate.js` in format and minify

**Comment by MaanooAk** [comment]
If don't want I could fix these issues in a separate pull request

**Comment by wellsjo** [comment]
@MaanooAk hey, sorry that was really sloppy, I'm fixing in a separate pr

**Pull request merged**